### PR TITLE
support JSON wihout a file actually being a .json file

### LINKED ISSUE #15: `keyFilename` has to be .json to be read as json
We currently mount the keyfile into the k8s pod as a secret. We couldn't understand why we kept getting `email is required`. It took way too much time to figure out that this library expects a json keyfile to be named json (https://github.com/stephenplusplus/google-auto-auth/blob/b87976f5726f7642ccaff70042b179311ac9b65b/index.js#L54-L61). Secrets in k8s are typically just a filename without the extension, so this lib apparently thought we passed a path to a pem file, and gave us odd stacktraces with JWT stuff.

Our diff to fix it, for posterity:
```diff
iff --git a/apps/asset-server/server.js b/apps/asset-server/server.js
index 646f74a..bbc99ba 100644
--- a/apps/asset-server/server.js
+++ b/apps/asset-server/server.js
@@ -1,6 +1,7 @@
 'use strict';
 
 const { join } = require('path');
+const { readFileSync } = require('fs');
 const express = require('express');
 const brakes = require('brakes');
 const AssetServer = require('asset-pipe-build-server');
@@ -38,7 +39,7 @@ if (config.ASSETS_SINK === 'fs') {
 } else if (config.ASSETS_SINK === 'gcs') {
     sink = new AssetSinkGCS({
         projectId: config.GCS_PROJECT_ID,
-        keyFilename: config.GCS_KEYFILE_LOCATION,
+        credentials: JSON.parse(readFileSync(config.GCS_KEYFILE_LOCATION, 'utf8')),
     }, 'podium-assets');
 } else {
     logger.error('Asset-pipe sink is not defined');
```

The point of this issue: Could either the docs be clarified about this implementation detail, or separate options for json keyfile and pem files be added?

As a bonus: if `keyFilename` points to a non-existent path, the same error (`email is required`) is thrown at the time of trying to talk to the API. IMO, an error saying that the file doesn't exist should be thrown during initialization 😄

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #15 

This will:

- Return an error if a provided `keyFile` or `keyFilename` cannot be found on disk
- Check if the contents of the provided keyfile can be parsed as JSON
  - if so, assume it is JSON
  - if not, assume it is PEM

### CODE REVIEW

**Comment by stephenplusplus** [comment]
@SimenB would you mind taking a look?

File: index.js [hunk 1]
```diff
@@ -51,19 +52,26 @@ Auth.prototype.getAuthClient = function (callback) {
...
-      var json = config.credentials;
+    if (config.credentials) {
+      googleAuth.fromJSON(config.credentials, addScope);
+    } else if (keyFile) {
+      keyFile = path.resolve(process.cwd(), keyFile);
 
-      if (!json) {
-        json = require(path.resolve(process.cwd(), keyFile));
-      }
+      fs.readFile(keyFile, function (err, contents) {
```
Comment (SimenB) [inline index.js]: Should this be `fs.readFile(keyFile, 'utf8', function (err, contents)`? Or does `JSON.parse` work on raw buffers?

https://nodejs.org/api/fs.html#fs_fs_readfile_file_options_callback

Comment (stephenplusplus) [inline index.js]: It actually works on raw buffers 👍

**Comment by stephenplusplus** [comment]
Thanks for the ideas and reviewing! Released as `0.6.1`

File: index.js (hunk 1 above)
Comment (SimenB) [inline index.js]: Yeah, I looked at the tests afterwards. Me gusta

Comment (theneva) [inline index.js]: I had no idea, nice!

**Comment by SimenB** [comment]
@stephenplusplus you've forgotten to push the commit updating package.json and the git tag 😄
https://github.com/stephenplusplus/google-auto-auth/blob/master/package.json#L3

**Comment by stephenplusplus** [comment]
Woops, done. Added you as a contributor to the repo since you've been such a help lately. No obligations to do anything, but your help is very appreciated and always welcome.

**Pull request merged**